Gopkg.toml

```diff
@@ -0,0 +1,109 @@
+[[constraint]] # Required to be compatible with `btcd`
```

Comment:
i would like to separate our deps and go-ethereum, what do you think about?

Reply:
I tried it, it doesn't seem to worth it.

1) If we have SHAs in the lock file (as we should) it isn't propagated to the main dependency. That makes the versions we use in `status-go` _slightly_ different from what `go-ethereum` is using. That is exactly what we want to avoid.

2) If we have SHAs as constraints in `go-ethereum/Gopkg.lock` (via manual labour), we will have dependency resolution issues if a dependency is used both in `status-go` and `go-ethereum` (`otto`, for instance).

So the most stable method I found is to keep all the dependencies in `Gopkg.toml` of `status-go`.

Reply:
i meant just to put our own deps on top of the file, then go-ethereum and then all of the go-ethereum deps. but maybe thats not worth it

Reply:
Ah, no, that is a good idea, I'll do that.
One thing to note is that we won't have much of our dependencies in the `toml` file anyways.
The idea of this `toml`/`lock` split that the toml file contains only _constrained_ dependencies. If we don't care about the version, we don't add this dep in there. So basically we will have:
1. `btcutils` (constrained because `btcd`)
2. `go-ethereum` (constrained because a custom source and a custom branch)
3. `geth` dependencies.

`dep` ensures that SHAs are in the `lock` file and the components aren't updated until you specifically ask for it.

Reply:
fixed in the latest push